homeassistant/components/heatzy/climate.py

```diff
@@ -0,0 +1,188 @@
+            ),
+        }
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes."""
+        return MODE_LIST
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode."""
+        if self.preset_mode == PRESET_NONE:
+            return HVAC_MODE_OFF
+        return HVAC_MODE_HEAT
+
+    @property
+    def preset_modes(self):
+        """Return a list of available preset modes."""
+        return PRESET_LIST
+
+    async def async_set_hvac_mode(self, hvac_mode):
+        """Set new hvac mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.async_turn_off()
+        elif hvac_mode == HVAC_MODE_HEAT:
+            await self.async_turn_on()
+
+    async def async_turn_on(self):
+        """Turn device on."""
+        await self.async_set_preset_mode(PRESET_COMFORT)
+
+    async def async_turn_off(self):
+        """Turn device off."""
+        await self.async_set_preset_mode(PRESET_NONE)
+
+
+class HeatzyPiloteV1Thermostat(HeatzyThermostat):
+    """Heaty Pilote v1."""
+
+    HEATZY_TO_HA_STATE = {
+        "\u8212\u9002": PRESET_COMFORT,
+        "\u7ecf\u6d4e": PRESET_ECO,
+        "\u89e3\u51bb": PRESET_AWAY,
+        "\u505c\u6b62": PRESET_NONE,
+    }
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: [1, 1, 0],
+        PRESET_ECO: [1, 1, 1],
+        PRESET_AWAY: [1, 1, 2],
+        PRESET_NONE: [1, 1, 3],
+    }
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode, e.g., home, away, temp."""
+        return self.HEATZY_TO_HA_STATE.get(
+            self._coordinator.data[self.unique_id].get("attr", {}).get("mode")
+        )
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new preset mode."""
+        try:
+            await self.hass.async_add_executor_job(
+                self._coordinator.heatzy_client.control_device,
+                self.unique_id,
+                {"raw": self.HA_TO_HEATZY_STATE.get(preset_mode)},
+            )
+        except HeatzyException as error:
+            _LOGGER.error("Error to set preset mode : %s", error)
+        await self.coordinator.async_request_refresh()
+
+
+class HeatzyPiloteV2Thermostat(HeatzyThermostat):
+    """Heaty Pilote v2."""
+
+    # spell-checker:disable
+    HEATZY_TO_HA_STATE = {
+        "cft": PRESET_COMFORT,
+        "eco": PRESET_ECO,
+        "fro": PRESET_AWAY,
+        "stop": PRESET_NONE,
+    }
+
+    HA_TO_HEATZY_STATE = {
+        PRESET_COMFORT: "cft",
+        PRESET_ECO: "eco",
+        PRESET_AWAY: "fro",
+        PRESET_NONE: "stop",
+    }
+    # spell-checker:enable
+
+    @property
+    def preset_mode(self):
+        """Return the current preset mode, e.g., home, away, temp."""
+        return self.HEATZY_TO_HA_STATE.get(
+            self._coordinator.data[self.unique_id].get("attr", {}).get("mode")
+        )
+
+    async def async_set_preset_mode(self, preset_mode):
```

Comment:
The API is sync, so we can just use the sync API Home assistant offers here:

```suggestion
    def set_preset_mode(self, preset_mode):
```

And adjust the method accordingly (removing add_executor_job)